src/pspm_split_sessions.m

```diff
@@ -154,17 +154,12 @@
 % 2.2 Handle missing epochs
 if options.missing
     % makes sure the epochs are in seconds and not empty
-    [sts, missing] = pspm_get_timing('epochs', options.missing, 'seconds');
+    [sts, missing_time] = pspm_get_timing('epochs', options.missing, 'seconds');
     if sts < 0
         warning('ID:invalid_input', 'Could not load missing epochs.');
     end
     missingsr = 10000; % dummy sample rate, should be higher than data sampling rates (but no need to make it dynamic)
-    if any(missing > ininfos.duration)
-        warning('ID:invalid_input', 'Some missing epochs are outside data file.');
-        return
-    else
-        missing = round(missing*missingsr); % convert epochs in sec to datapoints
-    end
+    missing = pspm_time2index(missing_time, missingsr, ininfos.duration*missingsr); % convert epochs in sec to datapoints

```

Comment:
could the last argument lead to problems as it is not an integer? result might also not be the same as in line 163 where ´´´round´´´is used?

Reply:
Obviously I made a stupid mistake... let me think how to make `pspm_time2index` easier to use

Reply:
I think the problem is that the duration is converted from a floating point number to integer in two different places: once in pspm_time2index, and once in line 163 with "round". The result is not very resilient: imagine somebody changes pspm_time2indx but forgets to update the "round" in line 163.

I think it would be much better if pspm_time2index would accept only an integer index duration, and the same integer would be used in line 163.

Reply:
I understand the issue now... obviously it is not necessary to calculate the duration twice. I have corrected this issue in the new pull request.